src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs

```diff
@@ -70,19 +72,14 @@ internal abstract partial class AbstractLanguageService<TPackage, TLanguageServi
         /// </remarks>
         private bool _isSetUp;
 
-        protected AbstractLanguageService(
-            TPackage package)
+        protected AbstractLanguageService(TPackage package)
         {
-            this.Package = package;
+            _globalOptions = package.ComponentModel.GetService<IGlobalOptionService>();
```

Comment:
I guess.

Not sure why we have `Setup` instead of doing that work in the constructor.

```C#
_languageService = CreateLanguageService();
_languageService.Setup();
```

Reply:
I found this issue when working in `SetupNewTextView`, because in that scenario `GetServices()` has not been called yet. This was painful to discover, but there is at least one code path that doesn't do the extra initialization.